### DISCUSSION TITLE
Configuration for moving failed jobs into archive table

### DISCUSSION THREAD

**Discussion by klesgidis** [timgit/pg-boss#362] [Q&A]
Hi! Thanks a lot for all of your work done on this project.

The `archiveCompletedAfterSeconds` archives not only `completed` jobs, but also failed.

Is there a way to move `failed` jobs into `archive` table with a different configuration option?

My use case is that we need more time to handle `failed` jobs than the completed. (e.g. deploy a fix in production and then retry the job)

**Comment by timgit** [comment]
Yes, that sounds like a good use case for a new config option

**Reply by klesgidis** [reply]
Ok, I'll open a PR soon

**Reply by klesgidis** [reply]
Any suggestions on where this should be implemented? Or should I take it on my own?

**Reply by timgit** [reply]
`archiveFailedAfterSeconds` seems like a good choice.  If it's not set, it can default to `archiveCompletedAfterSeconds`

**Comment by klesgidis** [comment]
https://github.com/timgit/pg-boss/pull/364
